File: src/main/java/com/vc/medium/L280.scala
package com.vc.medium

object L280 {
  def wiggleSort(arr: Array[Int]): Unit = {
    (1 until arr.length).foreach(i => {
      if(
        (i % 2 == 1 && arr(i) < arr(i - 1)) ||
        (i % 2 == 0 && arr(i) > arr(i - 1))
      ) {
        val tmp = arr(i - 1)
        arr(i - 1) = arr(i)
        arr(i) = tmp
      }
    })
  }
}

File: src/main/java/com/vc/medium/L280.java
package com.vc.medium;

import java.util.Arrays;

class L280 {
    public void wiggleSort(int[] arr) {
        if(arr == null || arr.length == 0) return;

        int n = arr.length;
        Arrays.sort(arr);
        for(int i = 1; i + 1 < n; i++) {
            int temp = arr[i];
            arr[i] = arr[i + 1];
            arr[i + 1] = temp;
        }
    }
}
